### ISSUE TITLE
keyword_idents misses lifetime in several places

### ISSUE THREAD

**Issue by ehuss** [rust-lang/rust#133839]
The `keyword_idents` lint seems to miss the lifetime in a few places. Example:

```rust
#![warn(keyword_idents)]

pub struct S<'gen>(pub &'gen i32);

pub trait Visit {
    fn visit_path(&mut self) {}
}

impl<'gen> Visit for S<'gen> {}
```

The suggestions applied results in:

```diff
@@ -1,9 +1,9 @@
 #![warn(keyword_idents)]
 
-pub struct S<'gen>(pub &'gen i32);
+pub struct S<'r#gen>(pub &'gen i32);
 
 pub trait Visit {
     fn visit_path(&mut self) {}
 }
 
-impl<'gen> Visit for S<'gen> {}
+impl<'r#gen> Visit for S<'gen> {}
```

Notice that it missed the `'r#gen` update in the tuple struct definition, and the `for S<'gen>` position. I have not audited to see what other scenarios it is missing.

### Meta

```
rustc 1.85.0-nightly (c44b3d50f 2024-12-03)
binary: rustc
commit-hash: c44b3d50fea96a3e0417e8264c16ea21a0a3fca2
commit-date: 2024-12-03
host: aarch64-unknown-linux-gnu
release: 1.85.0-nightly
LLVM version: 19.1.4
```

**Comment by ehuss** [comment]
Another example is an APIT:

```rust
pub fn inner<'gen>(func: impl FnOnce() + 'gen) {}
```

It gets the function generic parameter, but not the generic parameter in the APIT.

**Comment by ehuss** [comment]
Closing as fixed by #133992.
